inference-engine/src/mkldnn_plugin/nodes/extract_image_patches.cpp

```diff
@@ -1,248 +1,485 @@
+        jpp.block_size = 1;
+
+        if (mayiuse(x64::avx512_common)) {
+            jpp.block_size = cpu_isa_traits<x64::avx512_common>::vlen / jpp.dtype_size;
+            eximpat_kernel.reset(new jit_eximpat_kernel<x64::avx512_common>(jpp));
+        } else if (mayiuse(x64::avx2)) {
+            jpp.block_size = cpu_isa_traits<x64::avx2>::vlen / jpp.dtype_size;
+            eximpat_kernel.reset(new jit_eximpat_kernel<x64::avx2>(jpp));
+        } else if (mayiuse(x64::sse41)) {
+            jpp.block_size = cpu_isa_traits<x64::sse41>::vlen / jpp.dtype_size;
+            eximpat_kernel.reset(new jit_eximpat_kernel<x64::sse41>(jpp));
+        }
 
-        parallel_nt(0, thread_body);
-    }
+        if (eximpat_kernel)
+            eximpat_kernel->create_ker();
 
-private:
-    std::vector<int64_t> _ksizes;
-    std::vector<int64_t> _strides;
-    std::vector<int64_t> _rates;
-    std::string _auto_pad;
+        LayerConfig config;
 
-    static const std::set<size_t> _supported_precisions_sizes;
-};
+        DataConfig inConfig;
+        inConfig.desc = inData->getTensorDesc();
+        config.inConfs.push_back(inConfig);
 
-const std::set<size_t> ExtractImagePatchesImpl::_supported_precisions_sizes = {1, 2, 4, 8};
+        DataConfig outConfig;
+        outConfig.desc = layer->outData[0]->getTensorDesc();
+        outConfig.desc.setPrecision(inConfig.desc.getPrecision());
+        outConfig.desc.setLayout(inConfig.desc.getLayout());
+        config.outConfs.push_back(outConfig);
+
+        config.dynBatchSupport = false;
+        confs.push_back(config);
+    } catch (InferenceEngine::Exception &ex) {
+        errorMsg = ex.what();
+    }
+}
 
-REG_FACTORY_FOR(ExtractImagePatchesImpl, ExtractImagePatches);
 
+StatusCode ExtractImagePatchesImpl::execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs, ResponseDesc *resp) noexcept {
+    const char *src_data = inputs[0]->cbuffer().as<const char *>() +
+            inputs[0]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+    char *dst_data = outputs[0]->buffer().as<char *>() +
+            outputs[0]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+    const int64_t dtype_size = inputs[0]->getTensorDesc().getPrecision().size();
+
+    const auto& inDims = inputs[0]->getTensorDesc().getDims();
+    const int64_t IC = inDims[1];
+    const int64_t IH = inDims[2];
+    const int64_t IW = inDims[3];
+
+    const auto& outDims = outputs[0]->getTensorDesc().getDims();
+    const int64_t OB = outDims[0];
+    const int64_t OH = outDims[2];
+    const int64_t OW = outDims[3];
+
+    const int64_t KH = _ksizes[0], KW = _ksizes[1];
+    const int64_t SH = _strides[0], SW = _strides[1];
+    const int64_t RH = _rates[0], RW = _rates[1];
+    const int64_t PT = _pad_top, PL = _pad_left;
+
+    const std::vector<int64_t> ostrides = {KH * KW * IC * OH * OW, KW * IC * OH * OW, IC * OH * OW, OH * OW};
```

Comment:
1) Isn't KH * KW * IC equals OC?
2) BlockingDesc should already contain precomputed strides

Reply:
It is. Please note however, that ostrides is not actually a complete stride, since OC is split into KH * KW * IC there. It is needed because the displacement is calculated for a given kh, kw and ic in the lambda function. Hence I thought that it is more clear to live KH * KW * IC instead of OC. Now I see that this is still confusing, thank you.

I rearranged this part a bit and used the getstrides() method to initialize istrides and ostrides, and also introduced ostrides_partial to emphasize its difference from ostrides.